pkg/cluster/logs/logs.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/kind/pkg/cluster/nodes"
+	"sigs.k8s.io/kind/pkg/docker"
+	"sigs.k8s.io/kind/pkg/exec"
+	"sigs.k8s.io/kind/pkg/util"
+)
+
+type errFn func() error
+
+// Collect collects logs related to / from the cluster nodes and the host
+// system to the specified directory
+func Collect(nodes []nodes.Node, dir string) error {
+	prefixedPath := func(path string) string {
+		return filepath.Join(dir, path)
+	}
+	// helper to run a cmd and write the output to path
+	execToPath := func(cmd exec.Cmd, path string) error {
+		realPath := prefixedPath(path)
+		os.MkdirAll(filepath.Dir(realPath), os.ModePerm)
+		f, err := os.Create(realPath)
```

Comment:
When does `f` get closed?

Reply:
d'oh!

.. when the binary exists and loses all descriptors? :this_is_fine:

Reply:
😅 not 100% certain what happens there, but the godoc doesn't seem to specify calling Close is required 🙄 https://golang.org/pkg/os/#File.Close